index.js

```diff
@@ -44,9 +47,33 @@ class HyperDHT extends DHT {
       query: onpeers
     })
     this.once('close', () => {
+      clearTimeout(this._adaptiveTimeout)
       this._peers.destroy()
       this._store.clear()
     })
+
+    this._adaptiveTimeout = null
+
+    if (opts.adaptive) {
+      if (this.ephemeral !== true) {
+        this.destroy()
+        throw Error('adaptive mode can only applied when ephemeral: true')
+      }
+      this.once('ready', () => {
+        this._adaptiveTimeout = setTimeout(() => {
```

Comment:
we should move this to use guard-timeout if trivial now :) otherwise we can do so later

Reply:
added